src/store/store.ts

```diff
@@ -0,0 +1,144 @@
+import { IUser, IProduct, IOrder, IDecodedUser } from "../services/api/types";
+import { makeAutoObservable } from "mobx";
+import jwtDecode from "jwt-decode";
+import Cookies from "js-cookie";
+import axios from "axios";
+import {
+  fetchOrders,
+  fetchProducts,
+  login,
+  logout,
+  registration,
+} from "../services/api/AuthService";
+import { API_URL } from "../http";
+
+export default class Store {
+  user = {} as IUser;
+  isAuth = false;
+  products = [] as IProduct[];
+  orders = [] as IOrder[];
```

Comment:
It is not very good to write types as you did.
Try (declaring an object may require to enter some default data):
```suggestion
  user: IUser = {};
  isAuth = false;
  products: IProduct[] = [];
  orders: IOrder[] = [];
```